quadriga/src/main/java/edu/asu/diging/quadriga/core/service/impl/AsyncPatternProcessorImpl.java

```diff
@@ -0,0 +1,108 @@
+import edu.asu.diging.quadriga.core.service.AsyncPatternProcessor;
+import edu.asu.diging.quadriga.core.service.MappedTripleGroupService;
+import edu.asu.diging.quadriga.core.service.MappedTripleService;
+import edu.asu.diging.quadriga.core.service.PatternFinder;
+import edu.asu.diging.quadriga.core.service.PatternMapper;
+import edu.asu.diging.quadriga.core.exceptions.JobNotFoundException;
+
+@Component
+public class AsyncPatternProcessorImpl implements AsyncPatternProcessor {
+
+    private Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private PatternFinder patternFinder;
+
+    @Autowired
+    private MappedTripleService mappedTripleService;
+
+    @Autowired
+    private MappedTripleGroupService mappedTripleGroupService;
+
+    @Autowired
+    private PatternMapper patternMapper;
+
+    @Autowired
+    private JobRepository jobRepository;
+
+    /* (non-Javadoc)
+     * @see edu.asu.diging.quadriga.core.service.AsyncPatternProcessor#processPattern(java.lang.String, java.lang.String, edu.asu.diging.quadriga.api.v1.model.PatternMapping, java.util.List)
+     */
+    @Override
+    @Async
+    public void processPattern(String jobId, String collectionId, PatternMapping patternMapping,
+            List<EventGraph> networks) throws JobNotFoundException {
+
+        Job job = jobRepository.findById(jobId).orElse(null);
+        PatternCreationEvent patternRoot = patternMapper.mapPattern(patternMapping);
+        
+        if (job == null) {
+            throw new JobNotFoundException();  
+        } 
+        
+        job.setStatus(JobStatus.PROCESSING);
+        jobRepository.save(job);
+        
+        MappedTripleGroup mappedTripleGroup;
+        try {
+            if (patternMapping.getMappedTripleGroupId() != null && !patternMapping.getMappedTripleGroupId().isEmpty()) {
+                mappedTripleGroup = mappedTripleGroupService.findByCollectionIdAndId(collectionId,
+                        patternMapping.getMappedTripleGroupId());
+                if (mappedTripleGroup == null) {
+                    mappedTripleGroup = new MappedTripleGroup();
+                    mappedTripleGroup.set_id(new ObjectId(patternMapping.getMappedTripleGroupId()));
+                    jobRepository.save(job);
+                }
+            } else {
+                mappedTripleGroup = mappedTripleGroupService.get(collectionId, MappedTripleType.CUSTOM_MAPPING);
+            }
+        } catch (InvalidObjectIdException | CollectionNotFoundException e) {
+            logger.error("Invalid triple group id {} or collection id {} for job {}",
+                    patternMapping.getMappedTripleGroupId(), collectionId, job.getId(), e);
+
+            job.setStatus(JobStatus.FAILURE);
+            jobRepository.save(job);
+            return;
+        }
+
+        for (EventGraph network : networks) {
+            List<Graph> extractedGraphs = patternFinder.findGraphsWithPattern(patternMapping.getMetadata(), patternRoot,
```

Comment:
is that method only finding graphs or also mapping them? It seems like it's also mapping them, in which case the method name is incorrect.